spec/parser_spec.lua

```diff
@@ -419,6 +419,19 @@ describe("Parser /", function()
             ]], "local function name has a ':'")
         end)
 
+        it("disallow too many function parameters", function ()
+           local params = ''
+           for i = 1, 200 do
+              params = params..'a'..i..', '
+           end
+           params = params..'a201'
+           assert_toplevel_error(string.format([[
+                function m.f(%s)
+                    return 1
+                end
+           ]], params), "too many parameters (limit is 200)")
```

Comment:
I'm very much in favor of generating the argument list programatically, instead of writing it by hand. However, in other tests that do this kind of thing we use string concatenation instead of string.format. For example, see the "maxlenfield" test cases in the execution tests.

    function m.f(]]..params..[[)